content/automate/ManageIQ/System/Request.class/__methods__/vm_retire_extend.rb

```diff
@@ -0,0 +1,64 @@
+#
+# Description: This method is used to add 14 days to retirement date when target
+# VM has a retires_on value and is not already retired
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Request
+        class VmRetireExtend
+          DEFAULT_EXTENSION_DAYS = 14
+          def initialize(handle = $evm)
+            @handle = handle
+          end
+
+          def main
+            validate
+            extend_retirement
+          end
+
+          private
+
+          def vm
+            @vm ||= @handle.root["vm"].tap do |vm|
+              if vm.nil?
+                @handle.log(:error, 'vm is nil')
+                raise 'ERROR - vm object not passed in'
+              end
+            end
+          end
+
+          def validate
+            if vm.retires_on.blank?
+              raise "ERROR - VM #{vm} has no retirement date - extension bypassed. No Action taken"
+            end
+
+            if vm.retired
+              raise "ERROR - VM #{vm} is already retired - extension bypassed. No Action taken"
+            end
+          end
+
+          def retirement_extend_days
+            @vm_retire_extend_days ||= @handle.object['vm_retire_extend_days'] || DEFAULT_EXTENSION_DAYS
+          end
+
+          def extend_retirement
+            @handle.log("info", "Number of days to extend: <#{retirement_extend_days}>")
+            @handle.log("info", "VM: <#{vm}> current retirement date is #{vm.retires_on}")
+            @handle.log("info", "Extending retirement <#{retirement_extend_days}> days for VM: <#{vm}>")
```

Comment:
@billfitzgerald0120 this still needs to be vm.name otherwise you are dealing with the whole vm object.